utils/pathfinder-probe/src/main.rs

```diff
@@ -0,0 +1,112 @@
+use anyhow::Context;
+use axum::{routing, Router};
+use metrics_exporter_prometheus::PrometheusBuilder;
+use std::{net::SocketAddr, time::Duration};
+
+// RUST_LOG=pathfinder_probe=debug ./target/release/pathfinder-probe 0.0.0.0:19999 https://alpha-mainnet.starknet.io http://127.0.0.1:9545 5
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::init();
+
+    let builder = PrometheusBuilder::new();
+    let handle = builder
+        .install_recorder()
+        .expect("failed to install recorder");
+
+    let setup = setup()?;
+    tracing::debug!(setup=?setup, "pathfinder-probe starting");
+
+    let listen_at: SocketAddr = setup.listen_at.parse().unwrap();
+    tracing::info!(server=?listen_at, "pathfinder-probe running");
+
+    tokio::spawn(async move {
+        loop {
+            if let Err(e) = tick(&setup).await {
+                tracing::error!(cause=?e, "Probe failed");
+            }
+            tokio::time::sleep(setup.poll_delay).await;
+        }
+    });
+
+    let app = Router::new().route("/metrics", routing::get(|| async move { handle.render() }));
+
+    axum::Server::bind(&listen_at)
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
+
+    Ok(())
+}
+
+#[derive(Debug)]
+struct Setup {
+    listen_at: String,
+    gateway_url: String,
+    pathfinder_url: String,
```

Comment:
You could consider early parsing of the first 3 config fields too :thinking:

Reply:
Done.